packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -769,10 +782,12 @@ export class AnchorSet implements Listenable<AnchorSetRootEvents>, AnchorLocator
 			notifyChildrenChanged(): void {
 				this.maybeWithNode(
 					(p) => {
+						assert(this.parentField !== undefined, "Must be in a field");
```

Comment:
Nit: might be worth being a bit more descriptive here for easier future debugging. "Must be in a field to..."